api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	Verified                 = "Verified"
+	AsExpected               = "AsExpected"
+	NotAsExpected            = "NotAsExpected"
+	MisConfiguredReason      = "MisConfigured"
+	RemovingReason           = "Removing"
+	PlatfromDestroyReason    = "Destroying"
+	HCIBeingConfiguredReason = "BeingConfigured"
+
+	// ProviderSecretConfigured indicates the state of the secret reference
+	ProviderSecretConfigured ConditionType = "ProviderSecretConfigured"
+
+	// PlatformConfigured indicates (if status is true) that the
+	// platform configuration specified for the platform provider has been applied
+	PlatformConfigured ConditionType = "PlatformInfrastructureConfigured"
+
+	// PlatformIAMConfigured indicates (if status is true) that the IAM is configured
+	PlatformIAMConfigured ConditionType = "PlatformIAMConfigured"
+)
+
+func init() {
+	SchemeBuilder.Register(&HostedClusterInfrastructure{}, &HostedClusterInfrastructureList{})
+}
+
+// HostedClusterInfrastructureSpec is the desired behavior of a HostedClusterInfrastructure.
+type HostedClusterInfrastructureSpec struct {
+
+	// InfraID is a globally unique identifier for the cluster. This identifier
+	// will be used to associate various cloud resources with the HostedCluster
+	// and its associated NodePools. If not specified the metadata.name for this
+	// resource is used. When specified, this value is used.
+	//
+	// +optional
+	// +immutable
+	InfraID string `json:"infraID,omitempty"`
+
+	// SubDomain is the identifier that is used between the *.apps and base-domain
+	// when building an ingress URL. This will also be the name of the cluster.
```

Comment:
what is the "name of the cluster"? hc.Name or something different?

Reply:
Yes, hc.Name.  In the param's for `create infra aws` we state its the name for the cluster. We don't state that it will become the subdomain for the clusters URL's.